src/machine/machine-dbus.c

```diff
@@ -875,6 +876,12 @@ int bus_machine_method_bind_mount(sd_bus_message *message, void *userdata, sd_bu
         if (r == 0)
                 return 1; /* Will call us back */
 
+        r = machine_get_uid_shift(m, &uid);
+        if (r < 0)
+                return r;
+        if (uid != 0)
+                return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED, "Can't bind mount on container with user namespacing applied.");
```

Comment:
Hmm, the most likely result of bind-mounting with shifted UIDs is that the container can only read the files, but not  write anything into it; this is certainly a common use case, and not particularly unsafe?